### PR TITLE
Add Java support

### PR DESCRIPTION
**Pull request by meta1203**
Add Java support for both full-file & snippet functionality.

One thing worth looking at is the fact that I added some extra regex parsing for Java specifically, as the compiler is very particular about how classes should be formatted. I figured that would be the better option as it would allow users to just Ctrl+C Ctrl+V from their IDE into Discord, but let me know if you want me to remove that before merging and I will.

### CODE REVIEW

File: discord/dockerfiles/java-Dockerfile [hunk 1]
```diff
@@ -0,0 +1,14 @@
+FROM openjdk:17
```
Comment (qbxt) [inline discord/dockerfiles/java-Dockerfile:1]: Does there happen to be an Alpine container for OpenJDK 17?

File: discord/help.py [hunk 2]
```diff
@@ -13,10 +13,11 @@ def __init__(self, bot):
         self.bash = "<:bash:831116845133332500>"
         self.js = "<:nodejs:834358450309300265>"
         self.cpp = "<:cpp:917582153544507442>"
+        self.java = "java" # TODO: get me a java discord emoji
```
Comment (qbxt) [inline discord/help.py:16]: I'll take care of this :)

File: launcher.template.sh [hunk 3]
```diff
@@ -1,7 +1,9 @@
...
 
 export EXECBOT_DISCORD_TOKEN=""
-export EXECBOT_ENV_TYPE=""
+export EXECBOT_ENV_TYPE="MASTER"
 export EXECBOT_LOGGING_WEBHOOK=""
 
-python main.py
+mkdir -p discord/playground
+
+cd discord && python main.py
```
Comment (qbxt) [inline launcher.template.sh:9]: This is a nice touch, thank you

**Review by qbxt — COMMENTED** [review]
One question in `java-Dockerfile`, otherwise LGTM. Thanks for the PR!

File: discord/dockerfiles/java-Dockerfile (hunk 1 above)
Comment (meta1203) [reply discord/dockerfiles/java-Dockerfile:1]: Not for OpenJDK 17. AFAIK, Alpine containers only exist for the JDK early access builds. (eg. JDK 19)

**Comment by qbxt** [comment]
screenshot is this intended behavior?

**Comment by meta1203** [comment]
> Is this intended behavior?
> 
> 

Due to Java's unique insistence on everything being in a class, the current code processing method is to check for a `main` function and either pass it directly to the compiler or wrap it in a main class/function. This can be fixed, but it'll require more than one regex pass (and therefore a rewrite of `code_processing.py`)

**Comment by qbxt** [comment]
Thanks for looking further into it, I think this is good as-is

I'll deploy this ASAP

**Pull request merged**